vertx-mysql-client/src/main/java/io/vertx/mysqlclient/impl/codec/MySQLDecoder.java

```diff
@@ -72,8 +72,18 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
   }
 
   private void decodePacket(ByteBuf payload, int payloadLength, int sequenceId) {
+    checkFireAndForgetCommands();
     CommandCodec<?, ?> ctx = inflight.peek();
     ctx.sequenceId = sequenceId + 1;
     ctx.decodePayload(payload, payloadLength);
+    checkFireAndForgetCommands();
+  }
+
+  private void checkFireAndForgetCommands() {
```

Comment:
this should be made static passing the inflight list as argument and reused in MySQLEncoder